src/Transients.cc

```diff
@@ -336,13 +336,23 @@ Transients::readers(const StoreEntry &e) const
 void
 Transients::markForUnlink(StoreEntry &e)
 {
-    unlink(e);
+    assert(e.key);
+    e.hasTransients() ? unlink(e) :
```

Comment:
> A tertiary operator imposes certain requirements on its second and third arguments, and ignoring its result may trigger compiler warnings.

In this particular case these 'requirements' should be satisfied and no warnings should be generated. Both arguments result is 'void' so the result of the operator is also 'void'.

Reply:
Yes, I know, but the return type(s) may change. The current `void` types are not a part of some natural interface invariant for those methods. The `if` statement would not be affected by such changes. A tertiary operator offers no advantages here, so I am puzzled why you are spending time to defend its use.

Reply:
>  I am puzzled why you are spending time to defend its use.

I am not going to defend this minor issue. In general (if everything is equal), I would prefer one line of the ternary operator to a clumsy equivalent four lines of if/else.

Reply:
> And (if the return types change), we would need to fix our if/else, prefixing methods with (void) methods ...

Not necessarily. The `if` statement will continue to work and read correctly if the method(s) gain a an optional return value. If the return value is required/important, then the `if` statement will need to be changed, but the changes are very likely to be much smaller in this case (because the two methods are too different to return the same kind of a required/important result).

In general, I prefer using separate lines for conditions and actions, which results in a one line difference between a tertiary operator and an `if` statement. Removing that extra `else` line is not worth the problems that tertiary operator brings.

AFAICT, the tertiary operator is meant for obtaining a result from either of the two different algorithms. The code which does not care about the result should not use the operator that does.

Reply:
Done at a40bbf6.